### PR TITLE
Support setting proxy url in client opts

### PR DESCRIPTION
**Pull request by sboschman**
prerequisite to support https://github.com/rancher/terraform-provider-rancher2/issues/982

### CODE REVIEW

**Comment by kkaempf** [comment]
Hmm, this disables `ProxyFromEnvironment` 

@sboschman what's wrong with passing the proxy via the environment ?

**Comment by sboschman** [comment]
> Hmm, this disables `ProxyFromEnvironment`
The idea is to use `ProxyFromEnvironment` as default [else block](https://github.com/rancher/norman/pull/453/files#diff-7db4fd30c2b01d2f37e474a7b480aba3f59b48c6606e9aa232169319319f6822R195), unless a proxy url is explicitly set in the ClientOpts.

The problem with passing the proxy via the env is described in the linked issue. It sets the same proxy for every Terraform provider as the env is picked up by golang internally. If the proxy is only for the Rancher provider, but not for any other provider used in the same terraform run you are stuck.

**Comment by snasovich** [comment]
Since the change is in Norman, assigning team responsible for this component instead. FYI @cbron @samjustus

File: clientbase/common.go
Comment (MbolotSuse) [inline clientbase/common.go:191]: Nit:

If we can wrap this error, that would provide more context on what specific part of the call failed. Something like:

```suggestion
			return result, fmt.Errorf("unable to parse url from url specified in options: %w", err)
```

This change isn't strictly necessary, but might help a bit with longer-term usability.

**Review by MbolotSuse — APPROVED** [review]
One small nitpick, but outside of that LGTM. Keep in mind that we will hold off on merging this until we have a second approval as well.

File: clientbase/common.go
Comment (sboschman) [reply clientbase/common.go]: Good point, returning the error now in the same way as `ProxyFromEnvironment` [does](https://github.com/golang/net/blob/master/http/httpproxy/proxy.go#L164) if it fails to parse the url.

File: clientbase/common.go
Comment (KevinJoiner) [inline clientbase/common.go:191]:
```suggestion
			return result, fmt.Errorf("invalid proxy address %q: %w", opts.ProxyURL, err)
```

**Review by KevinJoiner — CHANGES_REQUESTED** [review]

**Review by KevinJoiner — APPROVED** [review]

**Review by JonCrowther — APPROVED** [review]

**Pull request merged**
